packages/block-library/src/social-link/block.json

```diff
@@ -0,0 +1,15 @@
+{
+	"name": "core/social-link",
```

Comment:
I'm also wondering now that I noticed that the block's title is now `Social Icon`, should we change the name as well? :)

Reply:
When I started this PR I was pretty much on the `links` camp, but now I lean a bit towards `icon` too.

Reply:
@mcsf Why?

Reply:
Because I was initially considering the primary function of these blocks to be the linking — and this fit in with the possibility of declaring Social Links as a variation of Navigation or Buttons. But now I think we may compromise on functionality if we center the semantics of these blocks around just the linking.

Instead, their real purpose lies more in the appearance: the rendering of the links as icons. In the future, the icons may change looks, colours or shape, but they will remain icons. I think the user is best served if we are clear about this. If, instead, the primary purpose is linking, then other blocks are better suited (Navigation, Buttons, File, etc.).

Reply:
I have a diff that moves everything to Icons, but it's very large, so I'll open a separate PR to discuss this.

Reply:
Sure thing 💯

Reply:
The point of having social icon links is that they're links, not that they're icons, isn't it? From an accessibility standpoint, I question whether we should even have a block dedicated to textless buttons in the first place. Wouldn't that be sort of an anti-pattern?

Also, what if I want to have social links with visible text labels? Would I use the Buttons block for that?

Reply:
> The point of having social icon links is that they're links, not that they're icons, isn't it?

That's the subject part of the debate, I think.

> From an accessibility standpoint, I question whether we should even have a block dedicated to textless buttons in the first place. Wouldn't that be sort of an anti-pattern?

If done correctly, none of it should matter: icons should be as accessible as a regular text link, whether the reader cannot see, cannot use a cursor, etc.

> Also, what if I want to have social links with visible text labels? Would I use the Buttons block for that?

I could see either block type accommodating that use case.

I'll cc you on the new PR if you're interested in discussing this there.